src/IIIFPresentation/API/Features/Manifest/DlcsManifestCoordinator.cs

```diff
@@ -148,38 +171,56 @@ await dlcsApiClient.UpdateAssetManifest(dbManifest.CustomerId,
         canvasPaintingsToRemove.Select(cp => cp.AssetId?.ToString()).ToList(), OperationType.Remove,
     [dbManifest.Id], cancellationToken);
     
-    private async Task<(List<JObject> untrackedAssets, List<AssetId> dlcsAssetIds)> FindAssetsThatRequireAdditionalWork(List<JObject> assets, 
-        Models.Database.Collections.Manifest? dbManifest, int customerId, CancellationToken cancellationToken = default)
+    private async Task<AssetsWithAdditionalWork> FindAssetsThatRequireAdditionalWork(
+        List<JObject> assets, PresentationManifest presentationManifest, Models.Database.Collections.Manifest? dbManifest, 
+        int customerId, CancellationToken cancellationToken = default)
     {
-        var assetIds = assets.Select(a => a.GetAssetId(customerId));
-        List<CanvasPainting> assetsInDatabase = [];
-        List<AssetId> assetsTrackedElsewhere = [];
+        var assetIdsWithReingestion = presentationManifest.PaintedResources?.Select(pr =>
+                new AssetIdsWithReingestion(pr.Asset!.GetAssetId(customerId), pr.CanvasPainting?.Reingest ?? false))
+            .ToList() ?? [];
+
+        var (assetsInDatabase, trackedAssetsToReingest) = RetrieveTrackedAssets(customerId, assetIdsWithReingestion);
 
-        if (dbManifest != null)
+        var assetsTrackedElsewhere = assetsInDatabase.Where(a => a.ManifestId != dbManifest?.Id)
+            .Select(a => a.AssetId!).Distinct().ToList();
+
+        // all assets are tracked and not to be reingested
+        if (assetsInDatabase.Count == assetIdsWithReingestion.Count && !assetsTrackedElsewhere.Any() && !trackedAssetsToReingest.Any())
```

Comment:
For these two `.Any()` calls I'm seeing suggestion

> Prefer comparing 'Count' to 0 rather than using 'Any()', both for clarity and for performance

using extension method `.IsNullOrEmpty()` uses count while maintaining readability.